data/builtintype.h

```diff
@@ -24,16 +24,27 @@
 ****************************************************************************/
 #pragma once
 
+#include <QString>
 #include <QIcon>
+#include <QMap>
 
-#include "data/type.h"
+#include "type.h"
 
 namespace Asn1Acn {
 namespace Internal {
-namespace Icons {
+namespace Data {
 
-QIcon iconForType(Data::Type type);
+class BuiltinType : public Type
+{
+public:
+    BuiltinType(const QString &name);
```

Comment:
this string really hides an enum :(

Reply:
This is true, so do you think that it is better idea to add another inheritance level? This way Data::BuiltinType would become abstract, and classes which would represent different types  would be as follows (example for integer): 

```
class Integer : public BuiltinType
{
    public:
        Integer() = default;
        ~Integer() = default;

        QString label() const override { return QLatin1String("Integer"); }
        QIcon icon() const override { return createIcon("/path/to/icon/of/integer"); }
};

```

than in place where types are created (AstXmlParser) would be factory function 

```
std::unique_ptr<Data::Type> AstXmlParser::createType(const QString &type)
{
    if (type == QLatin1String("IntegerType"))
        return std::make_unique<Data::Type::Integer>();
    else if (type == QLatin1String("NullType"))
         return std::make_unique<Data::Type::Null>();
    .
    .
    .
    return nullptr;
}
```

?

Reply:
hmmm that would be the most OOP solution, but generates a lot of classes (still - we might keep them in single BuiltinTypes.cpp/h) Wouldn't some enum in BuiltIn type itself be sufficient?

yet, if you have energy for that - go with full OOP ;)

Reply:
Ok, so I will go with OOP solution. I believe that there will be some pay off from this approach in future :)